File: app/src/main/java/com/chatai/adr/api/GeminiApiService.java
package com.chatai.adr.api;

import com.chatai.adr.model.GeminiRequest;
import com.chatai.adr.model.GeminiResponse;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface GeminiApiService {

    @POST("v1beta/models/{model}:generateContent")
    Call<GeminiResponse> generateContent(
            @Path("model") String model,
            @Query("key") String apiKey,
            @Body GeminiRequest request
    );
}

File: app/src/main/java/com/chatai/adr/api/GeminiApiService.kt
package com.chatai.adr.api

import com.chatai.adr.model.GeminiRequest
import com.chatai.adr.model.GeminiResponse
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.POST
import retrofit2.http.Path
import retrofit2.http.Query

interface GeminiApiService {
    @POST("v1beta/models/{model}:generateContent")
    fun generateContent(
        @Path("model") model: String,
        @Query("key") apiKey: String,
        @Body request: GeminiRequest
    ): Call<GeminiResponse>
}
